pkgs/development/tools/gi-crystal/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, fetchFromGitHub
+, crystal
+, gobject-introspection
+}:
+crystal.buildCrystalPackage rec {
+  pname = "gi-crystal";
+  version = "0.17.0";
+
+  src = fetchFromGitHub {
+    owner = "hugopl";
+    repo = "gi-crystal";
+    rev = "v${version}";
+    hash = "sha256-DIH8L8P8lkWzzVUj1Tbf9oTUvu9X7OT66APyUHiDkYk=";
+  };
+
+  # Make sure gi-crystal picks up the name of the so or dylib and not the leading nix store path
+  # when the package name happens to start with “lib”.
+  patches = [ ./src.patch ./store-friendly-library-name.patch ];
+
+  nativeBuildInputs = [ gobject-introspection ];
+  buildTargets = [ "generator" ];
+
+  doCheck = false;
+  doInstallCheck = false;
```

Comment:
Please see if we can get tests working by not overwriting the env variable

Reply:
We can't, it doesn't work. The test rule in the `Makefile` simply starts generating bindings with custom typelib paths that do not have them, and fails.